ygen/codegen.go

```diff
@@ -363,13 +333,13 @@ func (cg *YANGCodeGenerator) GenerateGoCode(yangFiles, includePaths []string) (*
 
 	// enumTypeMap stores the map of the path to type.
 	enumTypeMap := map[string][]string{}
-	codegenErr := NewYANGCodeGeneratorError()
+	var codegenErr util.Errors
 	var structSnippets []string
 	for _, structName := range orderedStructNames {
 		structOut, errs := writeGoStruct(structNameMap[structName], goStructs, cg.state,
 			cg.Config.CompressOCPaths, cg.Config.GenerateJSONSchema, cg.Config.GoOptions)
 		if errs != nil {
-			codegenErr.Errors = append(codegenErr.Errors, errs...)
+			util.AppendErrs(codegenErr, errs)
```

Comment:
That also raises the question of whether some of these at least could be covered in tests.

Reply:
Thanks for catching this -- fix applied.

In terms of testing -- the issue is that we're trying to catch errors that are not possible to inject in this function since they depend on something being wrong in the input YANG or the previous processing stage. I added tests that catch all these top-level errors, but the testing that the right error is caught if a previous stage gives bad input is on the per function unit tests.

Reply:
The `nil` check can't get removed in a bunch of cases, since we also do a `continue`.